src/skan/csr.py

```diff
@@ -1001,6 +1002,76 @@ def make_degree_image(skeleton_image):
     return degree_image
 
 
+def _simplify_graph(graph):
```

Comment:
This is the problem @kushaangupta! This is a very very slow way of doing this. Plus, you're not using weights for the junction-to-junction distance.

skan already implicitly creates the edge list for the graph when using `summarize`. You can then do:

```python
from scipy.sparse import coo_matrix

...
table = summarize(skeleton)
src = np.asarray(table['node-id-src'])
dst = np.asarray(table['node-id-dst'])
distance = np.asarray(table['branch-distance'])

n = max(np.max(src), np.max(dst))

g1 = coo_matrix((distance, (src, dst)), shape=(n+1, n+1)).tocsr()
g = g1 + g1.T  # make non-directed
```

Reply:
You might also want to remap the indices before creating the COO, using `skimage.segmentation.relabel_sequential`, for example.

Reply:
Thanks @jni! I've made the changes as you said and it's finally ready to be called a speedup

Reply:
What's the current timing comparison on the retina image?

Reply:
56 seconds is down to 2.6 seconds

Reply:
@jni The code seems to fail for `skeleton0` which has pixel numbering starting from 0. As per the docstring on Skeleton.graph, isn't it supposed to be starting from 1 onwards?

Reply:
Ah, yes, the docstring is outdated, now since #143 the id corresponds to the raveled index of the pixel. So you'll need offset=0 when you relabel. I think that should fix it. 🤞

Reply:
I guess then `skimage.segmentation.relabel_sequential` is off the table as it would only accept strictly positive offset. I'll try to relabel using the approach in `skan.csr.pixel_graph` then

Reply:
Thanks, it solved the problem. But I was wondering since the central node we are finding through this fast method isn't exact same as the one found from the complete method, but a junction or terminal node closest to it.

To improve it, I was wondering about this approach: we can use the center found by the fast method as an initial seed and go along all of this neighbors to see which one is relatively more central (greedily) until we hit another junction or terminal (or maybe when we stop getting neighbors with  better centrality).

Reply:
I think it's better to just leave the approximation. These points don't look to be too far off from the junctions?

https://github.com/scikit-image/scikit-image/issues/6259

The code complexity to me is not worth the small benefit in finding a slightly more central node.

Reply:
You're right, but I was just considering the base cases of simpler skeletons with maybe no junction nodes. Maybe we can just add a corner case handler for that if no junction nodes exist in graph then don't simplify?

Reply:
Yup, that's not bad. I also am not sure what the behaviour is or should be for disconnected skeletons. You should check that (one of the test skeletons is disconnected). Possibly we return the center only of the largest connected component? (`scipy.sparse.csgraph.connected_components`)

Reply:
I added the base case. For multi skeletons, maybe an argument could be added. It could handle them using connected components if argument is `"single"`, else would compute individual centers & sholl plots for all?